lib/backend/k8s/syncer.go

```diff
@@ -1047,20 +1015,55 @@ func (syn *kubeSyncer) parseGlobalConfigEvent(e watch.Event) *model.KVPair {
 	return kvp
 }
 
+func (syn *kubeSyncer) parseSystemNetworkPolicyEvent(e watch.Event) *model.KVPair {
+	return syn.parseCustomK8sResourceEvent(e, resources.SystemNetworkPolicyConverter{}, "SystemNetworkPolicy")
+}
+
 func (syn *kubeSyncer) parseIPPoolEvent(e watch.Event) *model.KVPair {
-	log.Debug("Parsing IPPool watch event")
+	return syn.parseCustomK8sResourceEvent(e, resources.IPPoolConverter{}, "IPPool")
+}
+
+func (syn *kubeSyncer) parseCustomK8sResourceEvent(
+	e watch.Event,
+	converter resources.CustomK8sResourceConverter,
+	resourceType string,
+) *model.KVPair {
 	// First, check the event type.
-	tpr, ok := e.Object.(*thirdparty.IpPool)
+	logContext := log.WithFields(log.Fields{
+		"ResourceType": resourceType,
+		"EventType":    e.Type,
+	})
+	tpr, ok := e.Object.(resources.CustomK8sResource)
 	if !ok {
-		log.Panicf("Invalid IPPool event. Type: %s, Object: %+v", e.Type, e.Object)
+		logContext.Panicf("Invalid custom resource event. Object: %+v", e.Object)
 	}
 
-	// Convert the received IPPool into a KVPair.
-	kvp := resources.ThirdPartyToIPPool(tpr)
+	logContext = logContext.WithField("Name", tpr.GetObjectMeta().GetName())
+	logContext.Debug("Parsing watch event")
 
-	// For deletes, we need to nil out the Value part of the KVPair
-	if e.Type == watch.Deleted {
-		kvp.Value = nil
+	// Convert the received resource into a KVPair.
+	kvp, err := converter.ToKVPair(tpr)
+	if err == nil {
+		// For deletes, we need to nil out the Value part of the KVPair
+		if e.Type == watch.Deleted {
+			kvp.Value = nil
+		}
+		return kvp
 	}
-	return kvp
+
+	// Error converting resource.  Attempt to determine the Key and treat as
+	// a delete (Value will be nil).
+	logContext.WithError(err).Info("Error in resource data - may treat as delete")
+	key, err := converter.NameToKey(tpr.GetObjectMeta().GetName())
+	if err == nil {
+		log.WithField("Key", key).WithError(err).Error("Error in resource data, treating as deleted")
+		return &model.KVPair{
+			Key: key,
+		}
+	}
+
+	// Could not determine the Key from the resource name - all we can do is
+	// ignore this event.
+	logContext.WithError(err).Error("Error in resource data, ignoring event")
```

Comment:
urm... but also i missed a log where is should be logContext... So guilty as charged (just on the wrong log)